### PR TITLE
Simplify parseElementaryOperation in regards to special instructions

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: libsolidity/inlineasm/AsmParser.cpp [hunk 1]
```diff
@@ -260,6 +261,17 @@ assembly::Expression Parser::parseExpression()
 				boost::lexical_cast<string>(args) +
 				" arguments)"
 			));
+		// Disallow instructions returning multiple values and DUP/SWAP
+		if (
+			instructionInfo(instr.instruction).ret != 1 ||
+			isDupInstruction(instr.instruction) ||
+			isSwapInstruction(instr.instruction))
```
Comment (axic) [inline libsolidity/inlineasm/AsmParser.cpp]: Explicit dup/swap check here is not required as it is covered by `ret != 1`, but it still may be beneficial to have it explicitly for readability and potential future changes to `instructions.cpp`.

File: libsolidity/inlineasm/AsmParser.cpp [hunk 2]
```diff
@@ -247,10 +247,11 @@ assembly::ForLoop Parser::parseForLoop()
 assembly::Expression Parser::parseExpression()
 {
 	RecursionGuard recursionGuard(*this);
-	ElementaryOperation operation = parseElementaryOperation(true);
+	ElementaryOperation operation = parseElementaryOperation();
 	if (operation.type() == typeid(Instruction))
 	{
 		Instruction const& instr = boost::get<Instruction>(operation);
+		// Enforce functional notation is for instructions requiring multiple arguments.
```
Comment (chriseth) [inline libsolidity/inlineasm/AsmParser.cpp]: Please remove `is`

File: libsolidity/inlineasm/AsmParser.cpp (hunk 1 above)
Comment (chriseth) [reply libsolidity/inlineasm/AsmParser.cpp]: Yep, I agree! I think you have a suplus `)` here, though.

**Review by chriseth — APPROVED** [review]

**Comment by axic** [comment]
Fixed the error message to pass the tests.

**Pull request merged**